jetty-websocket/websocket-common/src/main/java/org/eclipse/jetty/websocket/common/io/FrameFlusher.java

```diff
@@ -311,20 +313,26 @@ public FrameFlusher(ByteBufferPool bufferPool, Generator generator, EndPoint end
 
     public void close()
     {
-        if (closed.compareAndSet(false,true))
+        List<FrameEntry> entries = null;
+        
+        try(Locker.Lock l = lock.lock())
         {
-            LOG.debug("{} closing {}",this);
-            EOFException eof = new EOFException("Connection has been closed locally");
-            flusher.failed(eof);
```

Comment:
This call would call the callbacks of any entries that had been partially processed inside the locked section of code.  This is asking for deadlocks etc.